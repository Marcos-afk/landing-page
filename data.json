{
  "data": [
    {
      "id": 1,
      "attributes": {
        "title": "Pages",
        "slug": "pages",
        "createdAt": "2022-06-22T21:08:38.452Z",
        "updatedAt": "2022-06-23T00:10:48.610Z",
        "publishedAt": "2022-06-23T00:10:48.482Z",
        "footer_text": "Feito com amor por Marcos Andr√© üß°",
        "sections": [
          {
            "id": 1,
            "__component": "section.section-two-columns",
            "title": "FIREFOX 85",
            "description": "To wrap up January, we are proud to bring you the release of Firefox 85.",
            "metadata": {
              "id": 1,
              "name": "home_page",
              "section_id": "homePage",
              "background": true
            },
            "image": {
              "data": {
                "id": 3,
                "attributes": {
                  "name": "javascript.svg",
                  "alternativeText": "javascript.svg",
                  "caption": "javascript.svg",
                  "width": null,
                  "height": null,
                  "formats": null,
                  "hash": "javascript_18b37d718b",
                  "ext": ".svg",
                  "mime": "image/svg+xml",
                  "size": 30.31,
                  "url": "https://res.cloudinary.com/dvkp9flm8/image/upload/v1655932011/javascript_18b37d718b.svg",
                  "previewUrl": null,
                  "provider": "cloudinary",
                  "provider_metadata": {
                    "public_id": "javascript_18b37d718b",
                    "resource_type": "image"
                  },
                  "createdAt": "2022-06-22T21:06:50.439Z",
                  "updatedAt": "2022-06-22T21:06:50.439Z"
                }
              }
            }
          },
          {
            "id": 2,
            "__component": "section.section-grid",
            "title": "MY GRID",
            "description": "Uma breve descri√ß√£o.",
            "metadata": {
              "id": 2,
              "name": "grid-one",
              "section_id": "gridOne",
              "background": true
            },
            "text_grid": [
              {
                "id": 4,
                "title": "Teste 1",
                "description": "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias."
              },
              {
                "id": 5,
                "title": "Teste 2",
                "description": "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias."
              },
              {
                "id": 6,
                "title": "Teste 3",
                "description": "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias."
              }
            ],
            "image_grid": []
          },
          {
            "id": 3,
            "__component": "section.section-grid",
            "title": "GALLERY",
            "description": "Uma breve descri√ß√£o.",
            "metadata": {
              "id": 3,
              "name": "gallery",
              "section_id": "gallery",
              "background": false
            },
            "text_grid": [],
            "image_grid": [
              {
                "id": 2,
                "image": {
                  "data": [
                    {
                      "id": 4,
                      "attributes": {
                        "name": "photo-1614232296132-8e2b98031ab2.jpg",
                        "alternativeText": "photo-1614232296132-8e2b98031ab2.jpg",
                        "caption": "photo-1614232296132-8e2b98031ab2.jpg",
                        "width": 360,
                        "height": 360,
                        "formats": {
                          "thumbnail": {
                            "ext": ".jpg",
                            "url": "https://res.cloudinary.com/dvkp9flm8/image/upload/v1655933012/thumbnail_photo_1614232296132_8e2b98031ab2_1ac4be2f3b.jpg",
                            "hash": "thumbnail_photo_1614232296132_8e2b98031ab2_1ac4be2f3b",
                            "mime": "image/jpeg",
                            "name": "thumbnail_photo-1614232296132-8e2b98031ab2.jpg",
                            "path": null,
                            "size": 4.08,
                            "width": 156,
                            "height": 156,
                            "provider_metadata": {
                              "public_id": "thumbnail_photo_1614232296132_8e2b98031ab2_1ac4be2f3b",
                              "resource_type": "image"
                            }
                          }
                        },
                        "hash": "photo_1614232296132_8e2b98031ab2_1ac4be2f3b",
                        "ext": ".jpg",
                        "mime": "image/jpeg",
                        "size": 16.44,
                        "url": "https://res.cloudinary.com/dvkp9flm8/image/upload/v1655933011/photo_1614232296132_8e2b98031ab2_1ac4be2f3b.jpg",
                        "previewUrl": null,
                        "provider": "cloudinary",
                        "provider_metadata": {
                          "public_id": "photo_1614232296132_8e2b98031ab2_1ac4be2f3b",
                          "resource_type": "image"
                        },
                        "createdAt": "2022-06-22T21:23:30.246Z",
                        "updatedAt": "2022-06-22T21:23:30.246Z"
                      }
                    }
                  ]
                }
              },
              {
                "id": 3,
                "image": {
                  "data": [
                    {
                      "id": 5,
                      "attributes": {
                        "name": "photo-1537200085645-4342dbcf0c14.jpg",
                        "alternativeText": "photo-1537200085645-4342dbcf0c14.jpg",
                        "caption": "photo-1537200085645-4342dbcf0c14.jpg",
                        "width": 360,
                        "height": 360,
                        "formats": {
                          "thumbnail": {
                            "ext": ".jpg",
                            "url": "https://res.cloudinary.com/dvkp9flm8/image/upload/v1655933074/thumbnail_photo_1537200085645_4342dbcf0c14_9e4386a863.jpg",
                            "hash": "thumbnail_photo_1537200085645_4342dbcf0c14_9e4386a863",
                            "mime": "image/jpeg",
                            "name": "thumbnail_photo-1537200085645-4342dbcf0c14.jpg",
                            "path": null,
                            "size": 6.69,
                            "width": 156,
                            "height": 156,
                            "provider_metadata": {
                              "public_id": "thumbnail_photo_1537200085645_4342dbcf0c14_9e4386a863",
                              "resource_type": "image"
                            }
                          }
                        },
                        "hash": "photo_1537200085645_4342dbcf0c14_9e4386a863",
                        "ext": ".jpg",
                        "mime": "image/jpeg",
                        "size": 25.07,
                        "url": "https://res.cloudinary.com/dvkp9flm8/image/upload/v1655933074/photo_1537200085645_4342dbcf0c14_9e4386a863.jpg",
                        "previewUrl": null,
                        "provider": "cloudinary",
                        "provider_metadata": {
                          "public_id": "photo_1537200085645_4342dbcf0c14_9e4386a863",
                          "resource_type": "image"
                        },
                        "createdAt": "2022-06-22T21:24:32.673Z",
                        "updatedAt": "2022-06-22T21:24:32.673Z"
                      }
                    }
                  ]
                }
              },
              {
                "id": 4,
                "image": {
                  "data": [
                    {
                      "id": 6,
                      "attributes": {
                        "name": "photo-1524010246915-297e8bb735e6.jpg",
                        "alternativeText": "photo-1524010246915-297e8bb735e6.jpg",
                        "caption": "photo-1524010246915-297e8bb735e6.jpg",
                        "width": 360,
                        "height": 360,
                        "formats": {
                          "thumbnail": {
                            "ext": ".jpg",
                            "url": "https://res.cloudinary.com/dvkp9flm8/image/upload/v1655933106/thumbnail_photo_1524010246915_297e8bb735e6_e3095f93f2.jpg",
                            "hash": "thumbnail_photo_1524010246915_297e8bb735e6_e3095f93f2",
                            "mime": "image/jpeg",
                            "name": "thumbnail_photo-1524010246915-297e8bb735e6.jpg",
                            "path": null,
                            "size": 8.56,
                            "width": 156,
                            "height": 156,
                            "provider_metadata": {
                              "public_id": "thumbnail_photo_1524010246915_297e8bb735e6_e3095f93f2",
                              "resource_type": "image"
                            }
                          }
                        },
                        "hash": "photo_1524010246915_297e8bb735e6_e3095f93f2",
                        "ext": ".jpg",
                        "mime": "image/jpeg",
                        "size": 26.69,
                        "url": "https://res.cloudinary.com/dvkp9flm8/image/upload/v1655933105/photo_1524010246915_297e8bb735e6_e3095f93f2.jpg",
                        "previewUrl": null,
                        "provider": "cloudinary",
                        "provider_metadata": {
                          "public_id": "photo_1524010246915_297e8bb735e6_e3095f93f2",
                          "resource_type": "image"
                        },
                        "createdAt": "2022-06-22T21:25:03.734Z",
                        "updatedAt": "2022-06-22T21:25:03.734Z"
                      }
                    }
                  ]
                }
              },
              {
                "id": 5,
                "image": {
                  "data": [
                    {
                      "id": 7,
                      "attributes": {
                        "name": "photo-1552327803-2c89bc3e1483.jpg",
                        "alternativeText": "photo-1552327803-2c89bc3e1483.jpg",
                        "caption": "photo-1552327803-2c89bc3e1483.jpg",
                        "width": 360,
                        "height": 360,
                        "formats": {
                          "thumbnail": {
                            "ext": ".jpg",
                            "url": "https://res.cloudinary.com/dvkp9flm8/image/upload/v1655933140/thumbnail_photo_1552327803_2c89bc3e1483_e952a6fa0c.jpg",
                            "hash": "thumbnail_photo_1552327803_2c89bc3e1483_e952a6fa0c",
                            "mime": "image/jpeg",
                            "name": "thumbnail_photo-1552327803-2c89bc3e1483.jpg",
                            "path": null,
                            "size": 1.58,
                            "width": 156,
                            "height": 156,
                            "provider_metadata": {
                              "public_id": "thumbnail_photo_1552327803_2c89bc3e1483_e952a6fa0c",
                              "resource_type": "image"
                            }
                          }
                        },
                        "hash": "photo_1552327803_2c89bc3e1483_e952a6fa0c",
                        "ext": ".jpg",
                        "mime": "image/jpeg",
                        "size": 5.46,
                        "url": "https://res.cloudinary.com/dvkp9flm8/image/upload/v1655933139/photo_1552327803_2c89bc3e1483_e952a6fa0c.jpg",
                        "previewUrl": null,
                        "provider": "cloudinary",
                        "provider_metadata": {
                          "public_id": "photo_1552327803_2c89bc3e1483_e952a6fa0c",
                          "resource_type": "image"
                        },
                        "createdAt": "2022-06-22T21:25:37.888Z",
                        "updatedAt": "2022-06-22T21:25:37.888Z"
                      }
                    }
                  ]
                }
              },
              {
                "id": 6,
                "image": {
                  "data": [
                    {
                      "id": 8,
                      "attributes": {
                        "name": "photo-1542611890-59dc3fb388fb.jpg",
                        "alternativeText": "photo-1542611890-59dc3fb388fb.jpg",
                        "caption": "photo-1542611890-59dc3fb388fb.jpg",
                        "width": 360,
                        "height": 360,
                        "formats": {
                          "thumbnail": {
                            "ext": ".jpg",
                            "url": "https://res.cloudinary.com/dvkp9flm8/image/upload/v1655933173/thumbnail_photo_1542611890_59dc3fb388fb_c7d7a15b16.jpg",
                            "hash": "thumbnail_photo_1542611890_59dc3fb388fb_c7d7a15b16",
                            "mime": "image/jpeg",
                            "name": "thumbnail_photo-1542611890-59dc3fb388fb.jpg",
                            "path": null,
                            "size": 5.06,
                            "width": 156,
                            "height": 156,
                            "provider_metadata": {
                              "public_id": "thumbnail_photo_1542611890_59dc3fb388fb_c7d7a15b16",
                              "resource_type": "image"
                            }
                          }
                        },
                        "hash": "photo_1542611890_59dc3fb388fb_c7d7a15b16",
                        "ext": ".jpg",
                        "mime": "image/jpeg",
                        "size": 15.26,
                        "url": "https://res.cloudinary.com/dvkp9flm8/image/upload/v1655933172/photo_1542611890_59dc3fb388fb_c7d7a15b16.jpg",
                        "previewUrl": null,
                        "provider": "cloudinary",
                        "provider_metadata": {
                          "public_id": "photo_1542611890_59dc3fb388fb_c7d7a15b16",
                          "resource_type": "image"
                        },
                        "createdAt": "2022-06-22T21:26:11.170Z",
                        "updatedAt": "2022-06-22T21:26:11.170Z"
                      }
                    }
                  ]
                }
              },
              {
                "id": 7,
                "image": {
                  "data": [
                    {
                      "id": 9,
                      "attributes": {
                        "name": "photo-1592675213417-9f0abf5966a5.jpg",
                        "alternativeText": "photo-1592675213417-9f0abf5966a5.jpg",
                        "caption": "photo-1592675213417-9f0abf5966a5.jpg",
                        "width": 360,
                        "height": 360,
                        "formats": {
                          "thumbnail": {
                            "ext": ".jpg",
                            "url": "https://res.cloudinary.com/dvkp9flm8/image/upload/v1655933205/thumbnail_photo_1592675213417_9f0abf5966a5_aec58408c0.jpg",
                            "hash": "thumbnail_photo_1592675213417_9f0abf5966a5_aec58408c0",
                            "mime": "image/jpeg",
                            "name": "thumbnail_photo-1592675213417-9f0abf5966a5.jpg",
                            "path": null,
                            "size": 4.58,
                            "width": 156,
                            "height": 156,
                            "provider_metadata": {
                              "public_id": "thumbnail_photo_1592675213417_9f0abf5966a5_aec58408c0",
                              "resource_type": "image"
                            }
                          }
                        },
                        "hash": "photo_1592675213417_9f0abf5966a5_aec58408c0",
                        "ext": ".jpg",
                        "mime": "image/jpeg",
                        "size": 14.83,
                        "url": "https://res.cloudinary.com/dvkp9flm8/image/upload/v1655933204/photo_1592675213417_9f0abf5966a5_aec58408c0.jpg",
                        "previewUrl": null,
                        "provider": "cloudinary",
                        "provider_metadata": {
                          "public_id": "photo_1592675213417_9f0abf5966a5_aec58408c0",
                          "resource_type": "image"
                        },
                        "createdAt": "2022-06-22T21:26:43.102Z",
                        "updatedAt": "2022-06-22T21:26:43.102Z"
                      }
                    }
                  ]
                }
              }
            ]
          },
          {
            "id": 4,
            "__component": "section.section-grid",
            "title": "MY GRID",
            "description": "Uma breve descri√ß√£o.",
            "metadata": {
              "id": 4,
              "name": "grid-two",
              "section_id": "gridTwo",
              "background": true
            },
            "text_grid": [
              {
                "id": 7,
                "title": "Teste 1",
                "description": "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias."
              },
              {
                "id": 8,
                "title": "Teste 2",
                "description": "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias."
              },
              {
                "id": 9,
                "title": "Teste 3",
                "description": "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias."
              }
            ],
            "image_grid": []
          },
          {
            "id": 1,
            "__component": "section.section-content",
            "title": "NEWS COVERAGE ",
            "content": "The release of Apple Silicon-based Macs at the end of last year generated a flurry of news coverage and some surprises at the machine‚Äôs performance.",
            "metadata": {
              "id": 5,
              "name": "intro",
              "section_id": "intro",
              "background": false
            }
          },
          {
            "id": 2,
            "__component": "section.section-content",
            "title": "PRICING",
            "content": "The release of Apple Silicon-based Macs at the end of last year generated a flurry of news coverage and some surprises at the machine‚Äôs performance.",
            "metadata": {
              "id": 6,
              "name": "pricing",
              "section_id": "pricing",
              "background": false
            }
          }
        ],
        "menu": {
          "id": 1,
          "logo_link": "#home",
          "logo_text": "Logo",
          "logo": {
            "data": null
          },
          "menu_link": [
            {
              "id": 1,
              "link_text": "intro",
              "url": "#intro",
              "open_in_new_tab": false
            },
            {
              "id": 2,
              "link_text": "grid-one",
              "url": "#grid-one",
              "open_in_new_tab": false
            },
            {
              "id": 3,
              "link_text": "gallery",
              "url": "#galerry",
              "open_in_new_tab": false
            },
            {
              "id": 4,
              "link_text": "grid-two",
              "url": "#grid-two",
              "open_in_new_tab": false
            },
            {
              "id": 5,
              "link_text": "pricing",
              "url": "#pricing",
              "open_in_new_tab": false
            },
            {
              "id": 6,
              "link_text": "contact",
              "url": "#contact",
              "open_in_new_tab": false
            }
          ]
        }
      }
    }
  ],
  "meta": {
    "pagination": {
      "page": 1,
      "pageSize": 25,
      "pageCount": 1,
      "total": 1
    }
  }
}
