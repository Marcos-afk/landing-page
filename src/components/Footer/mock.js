const data = {
  footerHtml: `<p><a href="https://www.linkedin.com/in/marcos-melo-94609b1b6/">Feito com ❤ por Marcos André</a></p>`,
};

export default data;
