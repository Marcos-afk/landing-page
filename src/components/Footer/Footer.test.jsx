import Footer from './';
import { renderTheme } from '../../styles/render-theme';
import { screen } from '@testing-library/react';
import data from './mock';

describe('<Footer />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Footer footerHtml={data.footerHtml} />);
    expect(screen.getByRole('link', { name: 'Feito com ❤ por Marcos André' })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
