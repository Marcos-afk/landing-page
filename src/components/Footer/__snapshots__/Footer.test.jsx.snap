// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render 1`] = `
.c4 {
  font-size: 2.4rem;
}

.c2 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  text-align: center;
  border-top: 0.1rem solid;
}

.c0 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .c3 {
  font-size: 1.6rem;
}

.c0 .c1 {
  padding-top: 0;
  padding-bottom: 0;
}

<div>
  <footer
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3 c4"
      >
        <p>
          <a
            href="https://www.linkedin.com/in/marcos-melo-94609b1b6/"
          >
            Feito com ❤ por Marcos André
          </a>
        </p>
      </div>
    </div>
  </footer>
</div>
`;
